### ISSUE TITLE
Prefilled Filename In File-Chooser

### ISSUE THREAD

**Issue by JAremko** [JAremko/ArcherBC2#9]
Currently, in our file operations, we don't have the ability to pre-specify the filename in the file chooser(it always starts empty). This feature could improve the user experience by providing a recommended filename, saving users' time, and reducing the possibility of typos or naming inconsistencies.

We'll have to rework SeeSaw's file chooser to support predefined(pre-filled) filenames.

**Comment by o-murphy** [comment]
Try to use [javax.swing.JFileChooser](https://javaswing.wordpress.com/2010/05/14/jfilechooser_using/)
it implements setting the initial file name

kinda this

```clojure
(import 'javax.swing.JFileChooser)

(defn save-as-chooser [*state]
  (let [file-chooser (JFileChooser.)]
    (doto file-chooser
      (.setFileSelectionMode JFileChooser/FILES_ONLY) ; Only allow file selection
      (.setDialogType JFileChooser/SAVE_DIALOG)      ; Set it as a save dialog
      (.setDialogTitle "Save File")                  ; Set the dialog title
      (.addChoosableFileFilter (javax.swing.filechooser.FileNameExtensionFilter. "Custom Files" ["a7p"])) ; Add file filter
      (.setSelectedFile (java.io.File. "default_filename.a7p"))) ; Set the initial file name

    ; Show the file chooser dialog and get the result
    (let [return-val (.showSaveDialog nil)]
      (if (= return-val JFileChooser/APPROVE_OPTION)
        (let [selected-file (.getSelectedFile file-chooser)]
          ; Call the save-as function with the selected file
          (save-as *state selected-file)) ; Assuming you have a modified save-as function to accept the file parameter
        (println "File selection canceled.")))))
```

**Comment by JAremko** [comment]
Thanks. Something like this should work. I'll  integrate it + #17

**Comment by JAremko** [comment]
Fixed with #17
